test/lint/connection_commands.rb

```diff
@@ -0,0 +1,189 @@
+    end
+
+    def test_client_reply
+      # Use the server commands interface that's known to work
+      assert_equal "OK", r.client(:reply, "ON")
+    end
+
+    def test_client_set_info
+      target_version "7.2" do
+        # Use the server commands interface that's known to work
+        assert_equal "OK", r.client(:set_info, "lib-name", "valkey-ruby")
+        assert_raises(Valkey::CommandError) do
+          r.client(:set_info, "invalid-attr", "value")
+        end
+      end
+    end
+
+    def test_client_unblock
+      # Use the server commands interface that's known to work
+      client_id = r.client(:id)
+      result = r.client(:unblock, client_id)
+      assert [0, 1].include?(result), "Unblock should return 0 or 1"
+    end
+
+    def test_client_no_evict
+      # Use the server commands interface that's known to work
+      assert_equal "OK", r.client_no_evict(:on)
+      assert_equal "OK", r.client_no_evict(:off)
+      assert_raises(Valkey::CommandError) do
+        r.client_no_evict(:invalid)
+      end
+    end
+
+    def test_client_no_touch
+      target_version "7.2" do
+        # Use the server commands interface that's known to work
+        assert_equal "OK", r.client_no_touch(:on)
+        assert_equal "OK", r.client_no_touch(:off)
+        assert_raises(Valkey::CommandError) do
+          r.client_no_touch(:invalid)
+        end
+      end
+    end
+
+    def test_client_getredir
+      redir = r.client_getredir
+      assert_kind_of Integer, redir
+    end
+
+    def test_hello_default
+      result = r.hello
+      # Backend returns array in current implementation
+      # TODO: Backend should convert RESP3 map to Ruby Hash
+      assert_kind_of Array, result
+      assert result.include?("server"), "HELLO response should contain server info"
+    end
+
+    def test_hello_with_version
+      result = r.hello(3)
+      # Backend returns array in current implementation
+      # TODO: Backend should convert RESP3 map to Ruby Hash
+      assert_kind_of Array, result
+      proto_index = result.index("proto")
+      assert_equal 3, result[proto_index + 1] if proto_index
+    end
+
+    def test_hello_with_setname
+      client_name = "hello_lint_test"
+      result = r.hello(3, setname: client_name)
+      # Backend returns array in current implementation
+      # TODO: Backend should convert RESP3 map to Ruby Hash
+      assert_kind_of Array, result
+      # In cluster mode, HELLO and CLIENT GETNAME might hit different nodes
+      # so we can't reliably assert the name was set
+      assert_equal client_name, r.client_get_name unless cluster_mode?
+    end
+
+    def test_reset
+      # Set some state
+      r.client_set_name("before_reset")
+      # In cluster mode, we can only use database 0
+      r.select(1) unless cluster_mode?
+      # Reset
+      result = r.reset
+      assert_equal "RESET", result
+      # State should be reset
+      assert_nil r.client_get_name
```

Comment:
This is failing for Ruby 3.1, 3.2, and 3.4, yet passes for 3 and 3.3.  Is this an intermittent timing issue?  Perhaps we could try adding a small sleep like `sleep(0.1)` before this final assertion?